OpenRA.Mods.Common/Activities/Turn.cs

```diff
@@ -18,28 +18,51 @@ namespace OpenRA.Mods.Common.Activities
 	public class Turn : Activity
 	{
 		readonly IDisabledTrait disablable;
+		readonly IFacing facing;
+		readonly Mobile mobile;
 		readonly int desiredFacing;
+		readonly bool setIsMoving;
 
-		public Turn(Actor self, int desiredFacing)
+		public Turn(Actor self, int desiredFacing, bool setIsMoving = false, bool isInterruptible = true)
 		{
 			disablable = self.TraitOrDefault<IMove>() as IDisabledTrait;
+			facing = self.Trait<IFacing>();
 			this.desiredFacing = desiredFacing;
+			this.setIsMoving = setIsMoving;
+			IsInterruptible = isInterruptible;
+
+			// This might look confusing, but the current implementation of Mobile is both IMove and IDisabledTrait,
+			// and this way we can save a separate Mobile trait look-up.
+			mobile = disablable as Mobile;
+		}
+
+		protected override void OnFirstRun(Actor self)
+		{
+			if (setIsMoving && mobile != null && !mobile.IsMoving)
+				mobile.IsMoving = true;
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (IsCanceled)
+			if (IsInterruptible && IsCanceled)
 				return NextActivity;
+
 			if (disablable != null && disablable.IsTraitDisabled)
 				return this;
 
-			var facing = self.Trait<IFacing>();
-
 			if (desiredFacing == facing.Facing)
 				return NextActivity;
+
 			facing.Facing = Util.TickFacing(facing.Facing, desiredFacing, facing.TurnSpeed);
 
 			return this;
 		}
+
+		protected override void OnLastRun(Actor self)
+		{
+			// If Mobile.IsMoving was set to 'true' OnFirstRun, we want to reset it to 'false' before the next tick.
```

Comment:
This comment isn't correct - if `Mobile.IsMoving` was true before the activity ran then `OnFirstRun` won't do anything and it will still be reset false here.